bindings/nodejs/src/secret_manager.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use iota_sdk_bindings_core::{
+    call_secret_manager_method as rust_call_secret_manager_method,
+    iota_sdk::client::secret::{SecretManager, SecretManagerDto},
+    Response, SecretManagerMethod,
+};
+use neon::prelude::*;
+
+pub struct SecretManagerMethodHandler {
+    channel: Channel,
+    secret_manager: SecretManager,
+}
+
+impl Finalize for SecretManagerMethodHandler {}
+
+impl SecretManagerMethodHandler {
+    fn new(channel: Channel, options: String) -> Arc<Self> {
+        let secret_manager_dto =
+            serde_json::from_str::<SecretManagerDto>(&options).expect("error initializing secret manager");
+        let secret_manager = SecretManager::try_from(&secret_manager_dto).expect("error initializing secret manager");
+
+        Arc::new(Self {
+            channel,
+            secret_manager,
+        })
+    }
+
+    async fn call_method(&self, method: String) -> (String, bool) {
+        match serde_json::from_str::<SecretManagerMethod>(&method) {
+            Ok(method) => {
+                let res = rust_call_secret_manager_method(&self.secret_manager, method).await;
+                let mut is_err = matches!(res, Response::Error(_) | Response::Panic(_));
+
+                let msg = match serde_json::to_string(&res) {
+                    Ok(msg) => msg,
+                    Err(e) => {
+                        is_err = true;
+                        serde_json::to_string(&Response::Error(e.into()))
+                            .expect("the response is generated manually, so unwrap is safe.")
```

Comment:
same here ... if it should ever panic due to a bit-flip or something than we shouldn't tell the user that it can't panic.

Reply:
Updated